internal/backend/connectionsinitsvc/svc.go

```diff
@@ -0,0 +1,157 @@
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"go.uber.org/fx"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/muxes"
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+type Svc struct{ svcs Svcs }
+
+type Svcs struct {
+	fx.In
+
+	Muxes        *muxes.Muxes
+	Connections  sdkservices.Connections
+	Integrations sdkservices.Integrations
+	Vars         sdkservices.Vars
+}
+
+func New(svcs Svcs) Svc { return Svc{svcs: svcs} }
+
+func Init(svcs Svcs) {
+	s := Svc{svcs}
+
+	// These paths should correspond to the ones enriched in the connection service.
+	svcs.Muxes.Auth.HandleFunc("POST /connections/{id}/test", s.testConnection)
+	svcs.Muxes.Auth.HandleFunc("POST /connections/{id}/refresh", s.refreshConnection)
+	svcs.Muxes.Auth.HandleFunc("GET /connections/{id}/init/{origin}", s.init)
+	svcs.Muxes.Auth.HandleFunc("GET /connections/{id}/postinit", s.postInit)
+	svcs.Muxes.Auth.HandleFunc("GET /connections/{id}/success", s.success)
+}
+
+func (s Svc) init(w http.ResponseWriter, r *http.Request) {
+	id, origin := r.PathValue("id"), r.PathValue("origin")
+
+	cid, err := sdktypes.StrictParseConnectionID(id)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	conn, err := s.svcs.Connections.Get(r.Context(), cid)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if !conn.IsValid() {
+		http.Error(w, "connection not found", http.StatusNotFound)
+		return
+	}
+
+	integ, err := s.svcs.Integrations.GetByID(r.Context(), conn.IntegrationID())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if !integ.IsValid() {
+		http.Error(w, "integration not found", http.StatusNotFound)
+		return
+	}
+
+	http.Redirect(w, r, fmt.Sprintf("%s?cid=%v&origin=%s", integ.ConnectionURL(), cid, origin), http.StatusFound)
+}
+
+// postInit is the last step in the connection initialization flow.
+// It saves the connection's data in the connection's scope, and
+// redirects the user to the last HTTP response, based on its origin
+// (local AK server / local VS Code extension / SaaS web UI).
+func (s Svc) postInit(w http.ResponseWriter, r *http.Request) {
+	vars, origin := r.URL.Query().Get("vars"), r.URL.Query().Get("origin")
+
+	var data []sdktypes.Var
+	if err := kittehs.DecodeURLData(vars, &data); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	cid, err := sdktypes.StrictParseConnectionID(r.PathValue("id"))
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	data = kittehs.Transform(data, func(v sdktypes.Var) sdktypes.Var {
+		return v.WithScopeID(sdktypes.NewVarScopeID(cid))
+	})
+
+	if err := s.svcs.Vars.Set(r.Context(), data...); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	var u string
+	switch origin {
```

Comment:
1. Switch with one option + default --> if/else
2. Please explain: how are CLI invocations of "ak connection init" handled now? (They used to have a different URL target that used the dashboard, what's the new behavior?)

Reply:
> 2. Please explain: how are CLI invocations of "ak connection init" handled now? (They used to have a different URL target that used the dashboard, what's the new behavior?)

the URL did not change

Reply:
> 1. Switch with one option + default --> if/else

i think a switch is a good fit here - we might add options, also this is very readable.

Reply:
This PR changes the CLI init flow because the case for it - that was the point of question 2. Since it's already deployed I'll open a bug on this - it creates a bad user experience in cloud.